contracts/dao/voting/lockdrop-vault/src/contract.rs

```diff
@@ -252,6 +312,91 @@ pub fn query_bonding_status(
     })
 }
 
+fn get_voting_power_for_address(
+    deps: Deps,
+    config: Config,
+    pool_type: PoolType,
+    address: String,
+    height: u64,
+) -> StdResult<Decimal256> {
+    get_voting_power(
+        deps,
+        config,
+        pool_type,
+        &LockdropQueryMsg::QueryUserLockupTotalAtHeight {
+            pool_type,
+            user_address: address,
+            height,
+        },
+        height,
+    )
+}
+
+fn get_voting_power_total(
+    deps: Deps,
+    config: Config,
+    pool_type: PoolType,
+    height: u64,
+) -> StdResult<Decimal256> {
+    get_voting_power(
+        deps,
+        config,
+        pool_type,
+        &LockdropQueryMsg::QueryLockupTotalAtHeight { pool_type, height },
+        height,
+    )
+}
+
+fn get_voting_power(
+    deps: Deps,
+    config: Config,
+    pool_type: PoolType,
+    msg: &LockdropQueryMsg,
+    height: u64,
+) -> StdResult<Decimal256> {
+    let lp_tokens: Option<Uint128> = deps
+        .querier
+        .query_wasm_smart(config.lockdrop_contract, msg)?;
+
+    let lp_tokens = Decimal256::new(Uint256::from(lp_tokens.unwrap_or_default()));
+
+    if lp_tokens.is_zero() {
+        return Ok(Decimal256::zero());
+    }
+
+    let asset_denom = match pool_type {
+        PoolType::ATOM => "uatom",
+        PoolType::USDC => "usdc",
+    };
```

Comment:
During TGE we have 2 pool created - NTRN/USDC and NTRN/ATOM.
one oracle tracks one pool. So, to calculate power of both LPatom and LPusdc you have either configure 2 oracles or use one lockdrop_vault per denom.